pkg/controllers/tagging/metrics.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tagging
+
+import (
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"sync"
+)
+
+var register sync.Once
+
+var (
+	workItemLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
+			Name:           "tagging_workitem_latency",
+			Help:           "workitem latency",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"workqueue"})
+
+	workItemError = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name:           "tagging_workitem_error",
+			Help:           "workitem error",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{"workqueue"})
+)
+
+// registerMetrics registers tagging-controller metrics.
+func registerMetrics() {
+	register.Do(func() {
+		legacyregistry.MustRegister(workItemLatency)
+		legacyregistry.MustRegister(workItemError)
+	})
+}
+
+func recordWorkitemMetrics(actionName string, timeTaken float64, err error) {
+	if err != nil {
+		workItemError.With(metrics.Labels{"workqueue": actionName}).Inc()
```

Comment:
Have a question around what the dimensions for this metric is going to look like - are we able to query this metric using actionName as well? Because we seem to be passing strings with node names wherever we're calling this function so need to understand if that makes sense or if we need actionName to be somewhat static.

Reply:
That's a great call out, as I'm based this change off of [aws_metrics.go](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws_metrics.go), I assume the dimensions for this are similar to [this](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws_metrics.go#L54). I'll check in CW and see how they are being produced. However, I think you're right as the usage in [aws.go](https://github.com/kubernetes/cloud-provider-aws/blob/master/pkg/providers/v1/aws.go#L1030) is also somewhat static.